### PR TITLE
Update reference tests and support fine grained epoch processing tests

### PR DESCRIPTION
**Pull request by ajsutton**
## PR Description
Update the reference tests to v1.1.0-alpha.2.  This requires support for finer grained epoch processing tests (`process_final_updates` has been split up) and for loading SSZ data with snappy.

The Altona reference tests are currently completely ignored as we only run phase 0 tests, but this will be the first step towards being able to selectively enable them.

## Fixed Issue(s)
#3654 

## Documentation

- [x] I thought about documentation and added the `documentation` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

**Review by rolfyone — APPROVED** [review]
LGTM

File: build.gradle [hunk 1]
```diff
@@ -699,22 +696,6 @@ def getCheckedOutGitCommitHash() {
...
-  reports {
-    xml.enabled true
-    csv.enabled true
-    html.destination file("build/reports/jacocoHtml")
-  }
-  onlyIf = { true }
-  doFirst {
-    executionData = files(executionData.findAll { it.exists() })
-  }
-}
```
Comment (ajsutton) [inline build.gradle:716]: Removed jacoco because we don't use it and it was causing tests to fail on java 15 for me.

File: build.gradle [hunk 2]
```diff
@@ -439,7 +436,7 @@ subprojects {
     }
     referenceTest {
       java {
-        srcDir file('src/referenceTest/generated')
+        srcDir file('src/referenceTest/generated_tests')
```
Comment (ajsutton) [inline build.gradle:440]: This makes IntelliJ a lot happier by not having production code in a test source tree.

File: eth-reference-tests/src/referenceTest/java/tech/pegasys/teku/reference/phase0/forkchoice/ForkChoiceTestExecutor.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+ */
+
+package tech.pegasys.teku.reference.phase0.forkchoice;
+
+import com.google.common.collect.ImmutableMap;
+import tech.pegasys.teku.reference.phase0.TestExecutor;
+
+public class ForkChoiceTestExecutor {
+  public static final ImmutableMap<String, TestExecutor> FORK_CHOICE_TEST_TYPES =
+      ImmutableMap.of("fork_choice/get_head", TestExecutor.IGNORE_TESTS);
```
Comment (ajsutton) [inline eth-reference-tests/src/referenceTest/java/tech/pegasys/teku/reference/phase0/forkchoice/ForkChoiceTestExecutor.java:21]: Currently fork choice reference tests are ignored.  We're running the original fork choice tests as an integration test.  Will need to add support for the reference test format at some point.

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/statetransition/epoch/AbstractEpochProcessor.java [hunk 4]
```diff
@@ -77,25 +75,26 @@ private static void applyDeltas(final MutableBeaconState state, final Deltas att
...
+    processJustificationAndFinalization(state, validatorStatuses.getTotalBalances());
+    processRewardsAndPenalties(state, validatorStatuses);
+    processRegistryUpdates(state, validatorStatuses.getStatuses());
+    processSlashings(state, validatorStatuses.getTotalBalances().getCurrentEpoch());
+    processEth1DataReset(state);
+    processEffectiveBalanceUpdates(state);
+    processSlashingsReset(state);
+    processRandaoMixesReset(state);
+    processHistoricalRootsUpdate(state);
+    processParticipationUpdates(state);
```
Comment (ajsutton) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/statetransition/epoch/AbstractEpochProcessor.java]: v1.1 of the spec split up `processFinalUpdates` for both phase 0 and Altair so mirror that here.

**Pull request merged**